src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs

```diff
@@ -154,18 +154,44 @@ public StreamWriter(string path, bool append, Encoding encoding)
         {
         }
 
+#if NET6_0_OR_GREATER
+        public StreamWriter(string path, FileStreamOptions options)
+            : this(path, UTF8NoBOM, options)
+        {
+        }
+
+        public StreamWriter(string path, Encoding encoding, FileStreamOptions options)
+            : this(ValidateArgsAndOpenPath(path, encoding, options), encoding, DefaultFileStreamBufferSize)
+        {
+        }
+
+        private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, FileStreamOptions options)
+        {
+            ValidateArgs(path, encoding);
+            if (options == null)
+                throw new ArgumentNullException(nameof(options));
+
+            return new FileStream(path, options);
+        }
+#endif
```

Comment:
`StreamWriter` is not included in [Microsoft.IO.Redist.csproj](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.IO.Redist/src/Microsoft.IO.Redist.csproj), there is no need for the `#if` here

```suggestion
        public StreamWriter(string path, FileStreamOptions options)
            : this(path, UTF8NoBOM, options)
        {
        }

        public StreamWriter(string path, Encoding encoding, FileStreamOptions options)
            : this(ValidateArgsAndOpenPath(path, encoding, options), encoding, DefaultFileStreamBufferSize)
        {
        }

        private static Stream ValidateArgsAndOpenPath(string path, Encoding encoding, FileStreamOptions options)
        {
            ValidateArgs(path, encoding);
            if (options == null)
                throw new ArgumentNullException(nameof(options));

            return new FileStream(path, options);
        }
```